src/components/Onboarding/OnboardingContext.tsx

```diff
@@ -0,0 +1,197 @@
+  },
+  4: {
+    step: "Edit your mentor profile structure",
+    route: baseRoute + "mentors/edit-profile",
+  },
+  5: {
+    step: "Edit your mentee profile structure",
+    route: baseRoute + "mentees/edit-profile",
+  },
+});
+
+const MentorOnboardingText = (baseRoute: string) => ({
+  1: {
+    step: "Fill out your profile",
+    route: baseRoute + "edit-profile",
+  },
+  2: {
+    step: "Set your availability",
+    route: baseRoute + "availability",
+  },
+});
+
+const MenteeOnboardingText = (baseRoute: string) => ({
+  1: {
+    step: "Fill out your profile",
+    route: baseRoute + "edit-profile",
+  },
+  2: {
+    step: "Browse through available mentors",
+    route: baseRoute + "mentors",
+  },
+});
+
+const authLevelToText = (authLevel: AuthorizationLevel) => {
+  switch (authLevel) {
+    case AuthorizationLevel.Admin:
+      return AdminOnboardingText;
+    case AuthorizationLevel.Mentor:
+      return MentorOnboardingText;
+    default:
+      return MenteeOnboardingText;
+  }
+};
+
+interface OnboardingContextType {
+  switchingRoutes: boolean;
+  OnboardingComponent: JSX.Element;
+}
+
+const OnboardingContext = createContext<OnboardingContextType | undefined>(
+  undefined
+);
+
+const useOnboardingProvider = () => {
+  const currentProfile = useCurrentProfile();
+  const [updateProfile] = useUpdateProfileMutation({
+    refetchQueries: ["getMyUser"],
+  });
+
+  const authorizationLevel = useAuthorizationLevel();
+  const [loading, setLoading] = useState(false);
+  const [currentStep, setCurrentStep] = useState<number>(1);
+  const router = useRouter();
+
+  const MAX_STEPS = authorizationLevelToMaxSteps(authorizationLevel);
+  const baseRoute = `/program/${router.query.slug}/${router.query.profileRoute}/`;
+
+  const onFinish = () => {
+    const updateProfileInput: UpdateProfileInput = {
+      ...currentProfile.currentProfile,
+      showOnboarding: false,
+    };
+    updateProfile({
+      variables: {
+        profileId: currentProfile.currentProfile!.profileId,
+        data: updateProfileInput,
+      },
+    })
+      .then(() => {
+        currentProfile.refetchCurrentProfile!();
+        LocalStorage.delete("Onboarding Step");
+      })
+      .catch((err) => console.error(err));
+  };
+
+  const props: OnboardingProps = {
+    currentStep,
+    setCurrentStep: (num: number) => {
+      LocalStorage.set("Onboarding Step", num);
+      setCurrentStep(num);
+    },
+    loading,
+    setLoading,
+    onboardingText: authLevelToText(authorizationLevel)(baseRoute),
+    MAX_STEPS,
+    onFinish,
+  };
+
+  const onboardingStep = LocalStorage.get("Onboarding Step");
+  useEffect(() => {
+    if (onboardingStep && typeof onboardingStep == "number") {
+      setCurrentStep(onboardingStep);
+    }
+  }, []);
+
+  if (
+    authorizationLevel !== AuthorizationLevel.Admin &&
+    router.asPath !== props.onboardingText[currentStep]["route"] &&
+    onboardingStep &&
+    typeof onboardingStep == "number"
```

Comment:
So there are a couple of different routes that require different onboarding but if a user navigates outside of the onboarding page (manually), then we have a way to counteract it because we check to see whether the user is still going through onboarding and if they are, we navigate them back to the designated onboarding step. The if statement, I don't know why I put authorizationLevel !== AuthorizationLevel.Admin I think that was a mistake, but the other conditions check for if the current path matches the step that you are supposed to be on and then it checks if the localstorage variable exists and is a number.

Reply:
I've corrected it :' )